dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -146,23 +146,23 @@ boolean isSystemClass(String className) {
   }
 
   /**
-   * Scans class files in the jar file and returns a {@link SymbolReferenceSet} populated with
-   * symbol references.
-   *
-   * @param jar absolute path to a jar file
+   * Returns symbol references (maps from classes to class symbols, method symbols and field
+   * symbols) found in the class path.
    */
-  SymbolReferenceSet scanSymbolReferencesInJar(Path jar) throws IOException {
-    checkArgument(jar.isAbsolute(), "The input jar file path is not an absolute path");
-    checkArgument(Files.isReadable(jar), "The input jar file path is not readable");
+  SymbolReferenceMaps findSymbolReferences() throws IOException {
+    SymbolReferenceMaps.Builder builder = new SymbolReferenceMaps.Builder();
 
-    SymbolReferenceSet.Builder symbolTableBuilder = SymbolReferenceSet.builder();
-    for (JavaClass javaClass : listClassesInJar(jar)) {
-      if (!isCompatibleClassFileVersion(javaClass)) {
-        continue;
+    for (Path jar : inputClassPath) {
+      for (JavaClass javaClass : listClassesInJar(jar)) {
+        if (!isCompatibleClassFileVersion(javaClass)) {
+          continue;
+        }
+        ClassFile source = new ClassFile(jar, javaClass.getClassName());
+        builder.addAll(findSymbolReferences(source, javaClass));
```

Comment:
Hmm, I thought this would eliminate the need to pass two arguments but I guess not.

Reply:
Right. Symbol references needs source.